cpp/src/interop/from_arrow_device.cu

```diff
@@ -0,0 +1,409 @@
+      auto type              = arrow_to_cudf_type(&view);
+      auto [out_view, owned] = get_column(&view, child, type, false, stream, mr);
+      if (out_owned_cols.empty()) {
+        out_owned_cols = std::move(owned);
+      } else {
+        out_owned_cols.insert(std::end(out_owned_cols),
+                              std::make_move_iterator(std::begin(owned)),
+                              std::make_move_iterator(std::end(owned)));
+      }
+      return out_view;
+    });
+
+  return std::make_tuple<column_view, owned_columns_t>(
+    {type,
+     static_cast<size_type>(input->length),
+     nullptr,
+     reinterpret_cast<bitmask_type const*>(input->buffers[0]),
+     static_cast<size_type>(input->null_count),
+     static_cast<size_type>(input->offset),
+     std::move(children)},
+    std::move(out_owned_cols));
+}
+
+template <>
+std::tuple<column_view, owned_columns_t> dispatch_to_cudf_column::operator()<cudf::list_view>(
+  ArrowSchemaView* schema,
+  const ArrowArray* input,
+  data_type type,
+  bool skip_mask,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  auto offsets_view = column_view{data_type(type_id::INT32),
+                                  static_cast<size_type>(input->length) + 1,
+                                  input->buffers[1],
+                                  nullptr,
+                                  0,
+                                  static_cast<size_type>(input->offset)};
+
+  ArrowSchemaView child_schema_view;
+  NANOARROW_THROW_NOT_OK(
+    ArrowSchemaViewInit(&child_schema_view, schema->schema->children[0], nullptr));
+  auto child_type = arrow_to_cudf_type(&child_schema_view);
+  auto [child_view, owned] =
+    get_column(&child_schema_view, input->children[0], child_type, false, stream, mr);
+
+  return std::make_tuple<column_view, owned_columns_t>(
+    {type,
+     static_cast<size_type>(input->length),
+     nullptr,
+     reinterpret_cast<bitmask_type const*>(input->buffers[0]),
+     static_cast<size_type>(input->null_count),
+     static_cast<size_type>(input->offset),
+     {offsets_view, child_view}},
+    std::move(owned));
+}
+
+std::tuple<column_view, owned_columns_t> get_column(ArrowSchemaView* schema,
+                                                    const ArrowArray* input,
+                                                    data_type type,
+                                                    bool skip_mask,
+                                                    rmm::cuda_stream_view stream,
+                                                    rmm::mr::device_memory_resource* mr)
+{
+  return type.id() != type_id::EMPTY
+           ? std::move(type_dispatcher(
+               type, dispatch_to_cudf_column{}, schema, input, type, skip_mask, stream, mr))
+           : std::make_tuple<column_view, owned_columns_t>(get_empty_type_column(input->length),
+                                                           {});
+}
+
+}  // namespace
+
```

Comment:
Since you can't overload/specialize based solely on the return type, I chose to go with the table_view, if you need a single column then it is returned as a table_view with a single column, by which you can pull the `column_view` from it. I can't think of a use case offhand where it would be useful / better to make this templated or otherwise and create a `unique_column_view_t`. 

Is there a case you can think of that isn't easily serviced by just doing `unique_table_view->column(0)` ?